src/index.js

```diff
@@ -506,6 +528,55 @@ export default class AuthnWidget {
     }
   }
 
+  showDeviceManagementPopup(evt) {
+    // close other popups if any
+    var popupDivs = document.querySelectorAll("[id^='device-management-popup-frame']");
+    for (var i = 0; i < popupDivs.length; ++i) {
+      var popupDisplayStatus = popupDivs[i].style.display;
+      if (popupDisplayStatus === 'block') {
+        popupDivs[i].style.display = 'none';
+      }
+    }
+    let source = evt.currentTarget;
+    var deviceId = source.dataset['mfaSelectionKebabMenuContainer'];
+    var docId = 'device-management-popup-frame-' + deviceId;
+    document.querySelector("[id='" + CSS.escape(docId) + "']").style.display = 'block';
```

Comment:
for all of these query selectors, we are looking for an ID. I don't think that we will actually have multiple on the page, right? I think we can optimize with a getElementById instead

Reply:
Updated.